crates/primitives/src/revm/env.rs

```diff
@@ -359,6 +343,105 @@ fn fill_op_tx_env<T: AsRef<Transaction>>(tx_env: &mut TxEnv, transaction: T, env
         }
     }
 }
+
+/// Helper type to work with different transaction types when configuring the EVM env. This is
+/// the infallible version of `TryFillableTransaction`.
+pub trait FillableTransaction {
+    /// Returns the hash of the transaction.
+    fn hash(&self) -> TxHash;
+
+    /// Fill the transaction environment with the given transaction.
+    fn fill_tx_env(&self, tx_env: &mut TxEnv);
```

Comment:
I'm actually not sure if we want to keep this function, this would only be useful when reusing an existing TxEnv, but has some pitfalls (clearing fields).

I'd be okay with removing this function entirely

Reply:
agree, I'm going to get rid of `fill_tx_env`, it should be based on the transaction and not re-used